pkg/controller/daemon/daemoncontroller.go

```diff
@@ -899,11 +924,16 @@ func (dsc *DaemonSetsController) nodeShouldRunDaemonPod(node *v1.Node, ds *exten
 		return false, false, false, err
 	}
 
+	// Check if need to fire InsufficientResourceError
+	fireInsufficientResourceError := shouldFireInsufficientResourceError(reasons)
+
 	for _, r := range reasons {
 		glog.V(4).Infof("DaemonSet Predicates failed on node %s for ds '%s/%s' for reason: %v", node.Name, ds.ObjectMeta.Namespace, ds.ObjectMeta.Name, r.GetReason())
 		switch reason := r.(type) {
 		case *predicates.InsufficientResourceError:
-			dsc.eventRecorder.Eventf(ds, v1.EventTypeWarning, FailedPlacementReason, "failed to place pod on %q: %s", node.ObjectMeta.Name, reason.Error())
+			if fireInsufficientResourceError {
+				dsc.eventRecorder.Eventf(ds, v1.EventTypeWarning, FailedPlacementReason, "failed to place pod on %q: %s", node.ObjectMeta.Name, reason.Error())
+			}
 			shouldSchedule = false
```

Comment:
Why not just return `false, false, false` for intentional `PredicateFailureError`, and move the `InsufficientResourceError` case to after `PredicateFailureError`?